lib/std/multi_array_list.zig

```diff
@@ -252,6 +308,62 @@ pub fn MultiArrayList(comptime T: type) type {
             return index;
         }
 
+        /// Add `count` new elements at position `index`, which have undefined values.
+        pub fn addManyAt(self: *Self, allocator: Allocator, index: usize, count: usize) Allocator.Error!void {
+            const new_len = self.len + count;
+            const new_capacity = growCapacity(self.capacity, new_len);
+
+            if (self.capacity >= new_capacity)
+                return self.addManyAtAssumeCapacity(index, count);
+
+            // Avoid copying allocated by resizing in place.
+            const old_memory = self.allocatedBytes();
+            if (allocator.resize(old_memory, capacityInBytes(new_capacity))) {
+                self.capacity = new_capacity;
+                return self.addManyAtAssumeCapacity(index, count);
+            }
```

Comment:
I don't think your unit test coverage is sufficient because this logic is wrong. MultiArrayList uses only 1 allocation for all arrays, so if it is successfully resized, you still need to move all the arrays besides the first one.

I'm not convinced attempting a resize makes sense for this data structure, but you could measure the perf and find out if you want to try it.

Reply:
You're right, it's probably too complicated to resize the buffer. I will remove this branch then.